src/snapshot/download.js

```diff
@@ -27,13 +27,18 @@ function downloadImage(gd, opts) {
     // default to png
     opts.format = opts.format || 'png';
     
-    gd._snapshotInProgress = true;
-    var promise = toImage(gd, opts);
-
-    var filename = opts.filename || gd.fn || 'newplot';
-    filename += '.' + opts.format;
-
     return new Promise(function(resolve,reject){
+        if(gd._snapshotInProgress){
+          reject('Snapshotting is unavailable in Internet Explorer. ' +
```

Comment:
we should reject a `new Error` instead of just a string. See [example](https://github.com/plotly/plotly.js/blob/master/src/plots/geo/geo.js#L113-L118).